crates/client/src/client.rs

```diff
@@ -938,8 +938,10 @@ impl FuelClient {
     pub async fn submit_and_await_status<'a>(
         &'a self,
         tx: &'a Transaction,
+        allow_preconfirmation: bool,
```

Comment:
can we add a new function which has this preconf argument? submit_and_await_preconfirmation perhaps?

it makes the function call more clear

Reply:
I don't like the idea of having a new endpoint for each parameter we have. Because what if we had the same thing for `estimate_predicates` and you want to make a `submit_and_await_status` with both options, you have to make two calls ? but you can't because it's a submit.